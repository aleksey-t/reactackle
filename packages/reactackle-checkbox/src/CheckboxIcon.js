import React from 'react';
import PropTypes from 'prop-types';
import { withTooltip } from 'reactackle-tooltip';
import { withTheme } from 'reactackle-core';
import { CheckboxIconStyled } from './styles/CheckboxIconStyled';

const propTypes = {
  checked: PropTypes.bool,
  disabled: PropTypes.bool,
  tooltipText: PropTypes.string,
  theme: PropTypes.object.isRequired,
  Tooltip: PropTypes.func.isRequired,
  toggleTooltip: PropTypes.func.isRequired,
  showTooltip: PropTypes.func.isRequired,
  hideTooltip: PropTypes.func.isRequired,
};

const defaultProps = {
  checked: void 0,
  disabled: false,
  tooltipText: '',
};

const CheckboxIconComponent = props => {
  const Tooltip = props.Tooltip;

  const tooltip = props.tooltipText
    ? <Tooltip text={props.tooltipText} />
    : null;

  const tooltipManagementProps = props.tooltipText
    ? {
        onClick: props.toggleTooltip,
        onFocus: props.showTooltip,
        onBlur: props.hideTooltip,
        onMouseEnter: props.showTooltip,
        onMouseLeave: props.hideTooltip,
        tabIndex: 1,
      }
    : {};

  return (
    <CheckboxIconStyled
      checked={props.checked}
      disabled={props.disabled}
      {...tooltipManagementProps}
    >
      {props.theme.reactackle.components.checkbox.icon.src}
      {tooltip}
    </CheckboxIconStyled>
  );
};

CheckboxIconComponent.propTypes = propTypes;
CheckboxIconComponent.defaultProps = defaultProps;
CheckboxIconComponent.displayName = 'CheckboxIcon';

export default withTheme(withTooltip(CheckboxIconComponent, true));
