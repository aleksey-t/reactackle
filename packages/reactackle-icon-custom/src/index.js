import { iconCustomSizeMixin } from './iconCustomSizeMixin';
import { IconCustom } from './IconCustom';

export {
  IconCustom,
  iconCustomSizeMixin,
};
