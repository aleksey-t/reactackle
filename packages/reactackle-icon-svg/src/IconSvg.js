import React from 'react';
import PropTypes from 'prop-types';
import { registerDefaultComponentTheme } from 'reactackle-core';
import IconSvgStyled from './styles/IconSvgStyled';
import componentTheme from './styles/theme';
import IconDefault from './icons/IconDefault';

registerDefaultComponentTheme('iconSvg', componentTheme);

const propTypes = {
  /** Turns on/off icon's border */
  border: PropTypes.bool,
  /** Set icon's border width */
  borderWidth: PropTypes.bool,
  /** Make icon rounded */
  rounded: PropTypes.bool,
  /** Set icon size */
  size: PropTypes.oneOf(['custom', 'small', 'normal', 'large', 'xlarge']),
  /** Set icon's color scheme */
  colorScheme: PropTypes.string,
  /** Set icon's fill color */
  color: PropTypes.string,
  /** Set icon's background color */
  backgroundColor: PropTypes.string,
  /** Set icon's flipping plane */
  flip: PropTypes.string,
  /** Rotate icon by some degree */
  rotate: PropTypes.number,
  /** Icon component */
  src: PropTypes.object,
};

const defaultProps = {
  border: false,
  borderWidth: 0,
  rounded: false,
  size: 'normal',
  colorScheme: 'dark',
  color: '',
  backgroundColor: '',
  flip: 'none',
  rotate: 0,
  src: null,
};

export const IconSvg = ({
  src,
  children,
  backgroundColor,
  colorScheme,
  color,
  size,
  border,
  borderWidth,
  flip,
  rounded,
  rotate,
  ...props
}) => (
  <IconSvgStyled
    backgroundColor={backgroundColor}
    customColor={color}
    colorScheme={colorScheme}
    flip={flip}
    sizeKey={size}
    rotate={rotate}
    border={border}
    borderWidth={borderWidth}
    rounded={rounded}
    {...props}
  >
    {src || children || IconDefault}
  </IconSvgStyled>
);

IconSvg.propTypes = propTypes;
IconSvg.defaultProps = defaultProps;
IconSvg.displayName = 'IconSvg';
