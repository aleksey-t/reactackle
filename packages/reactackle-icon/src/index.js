import Icon from './Icon';

export { Icon };
export * from './iconPropType';
export * from './iconStyleMixin';
