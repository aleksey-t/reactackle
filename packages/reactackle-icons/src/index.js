import { registerDefaultComponentTheme } from 'reactackle-core';
import componentTheme from './styles/theme';

registerDefaultComponentTheme('icon', componentTheme);

export * from './IconCustom';
export * from './IconSvg';
export * from './iconStyleMixin';
export * from './iconsPropType';
