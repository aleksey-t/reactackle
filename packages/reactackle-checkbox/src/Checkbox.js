import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { noop, isUndef, registerDefaultComponentTheme } from 'reactackle-core';
import { TooltipIcon } from 'reactackle-tooltip-icon';
import { CheckboxStyled } from './styles/CheckboxStyled';
import { CheckboxElementStyled } from './styles/CheckboxElementStyled';
import { CheckboxInputStyled } from './styles/CheckboxInputStyled';
import { CheckboxLabelStyled } from './styles/CheckboxLabelStyled';
import { LabelElementStyled } from './styles/LabelElementStyled';
import { LabelTextStyled } from './styles/LabelTextStyled';
import { MessageStyled } from './styles/MessageStyled';
import CheckboxIcon from './CheckboxIcon';
import componentTheme from './styles/theme';

registerDefaultComponentTheme('checkbox', componentTheme);

const propTypes = {
  /**
   * Determines whether to check Checkbox by default
   */
  defaultChecked: PropTypes.bool,
  /**
   * Determines Checkbox checked state
   */
  checked: PropTypes.bool,
  /**
   * Determines whether to disable checkbox
   */
  disabled: PropTypes.bool,
  /**
   * Label of Checkbox
   */
  label: PropTypes.string,
  /**
   * Label Component
   */
  labelElement: PropTypes.element,
  /**
   * Add tooltip
   */
  tooltip: PropTypes.string,
  /**
   * Pass error message
   */
  errorMessage: PropTypes.string,
  /**
   * Specify function that will be called on Checkbox checked state change
   */
  onChange: PropTypes.func,
};

const defaultProps = {
  defaultChecked: false,
  checked: void 0,
  disabled: false,
  label: '',
  tooltip: '',
  errorMessage: '',
  labelElement: null,
  onChange: noop,
};

let nextId = 0;

export default class Checkbox extends Component {
  constructor(props) {
    super(props);

    this._inputId = `checkbox-${nextId++}`;

    this.state = {
      checked: isUndef(props.checked) ? props.defaultChecked : props.checked,
    };

    this._handleChange = this._handleChange.bind(this);
  }

  componentWillReceiveProps(nextProps) {
    if (
      nextProps.checked !== this.props.checked &&
      isUndef(nextProps.checked)
    ) {
      this.setState({
        checked: this.props.checked,
      });
    }
  }

  _handleChange(event) {
    const checked = event.target.checked;

    if (isUndef(this.props.checked)) {
      this.setState({
        checked,
      });
    }

    this.props.onChange({
      value: checked,
    });
  }

  render() {
    const checked = isUndef(this.props.checked)
      ? this.state.checked
      : this.props.checked;

    const tooltip =
      this.props.tooltip && (this.props.label || this.props.labelElement)
        ? <TooltipIcon text={this.props.tooltip} />
        : null;

    const label = this.props.label
      ? <span>
          <LabelTextStyled hasTooltip={!!this.props.tooltip}>
            {this.props.label}
          </LabelTextStyled>
          {tooltip}
        </span>
      : null;

    const labelElement = this.props.labelElement
      ? <LabelElementStyled hasTooltip={!!this.props.tooltip}>
          {this.props.labelElement}
          {tooltip &&
            <LabelTextStyled>
              {tooltip}
            </LabelTextStyled>}
        </LabelElementStyled>
      : null;

    const checkboxIcon =
      !this.props.label && !this.props.labelElement
        ? <CheckboxIcon
            disabled={this.props.disabled}
            checked={checked}
            tooltipText={this.props.tooltip}
          />
        : <CheckboxIcon
            disabled={this.props.disabled}
            checked={checked}
          />;

    const message = this.props.errorMessage
      ? <MessageStyled>
          {this.props.errorMessage}
        </MessageStyled>
      : null;

    return (
      <CheckboxStyled>
        <CheckboxElementStyled
          textFree={!this.props.label && !this.props.labelElement}
        >
          <CheckboxInputStyled
            id={this._inputId}
            type="checkbox"
            checked={checked}
            disabled={this.props.disabled}
            onChange={this._handleChange}
          />
          <CheckboxLabelStyled
            textFree={!this.props.label && !this.props.labelElement}
            disabled={this.props.disabled}
            checked={checked}
            htmlFor={this._inputId}
          >
            {checkboxIcon}
            {label}
            {labelElement}
          </CheckboxLabelStyled>
        </CheckboxElementStyled>
        {message}
      </CheckboxStyled>
    );
  }
}

Checkbox.propTypes = propTypes;
Checkbox.defaultProps = defaultProps;
Checkbox.displayName = 'Checkbox';
