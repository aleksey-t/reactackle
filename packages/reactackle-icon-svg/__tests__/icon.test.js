'use strict';

import React from 'react';
import renderer from 'react-test-renderer';
import { IconSvg } from '../src';

jest.mock('react-dom');

describe('<IconSvg/>', () => {
  test('renders correctly with default props', () => {
    const tree = renderer.create(
      <IconSvg />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('renders correctly with children prop', () => {
    const tree = renderer.create(
      <IconSvg>
        <svg viewBox="0 0 24 24" preserveAspectRatio="xMidYMid meet">
          <path d="M7,10l5,5l5-5H7z" />
        </svg>
      </IconSvg>,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('renders correctly with src prop', () => {
    const tree = renderer.create(
      <IconSvg
        src={
          <svg viewBox="0 0 24 24" preserveAspectRatio="xMidYMid meet">
            <path d="M7,10l5,5l5-5H7z" />
          </svg>
        }
      />,
  );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('renders correctly with border prop', () => {
    const tree = renderer.create(
      <IconSvg border />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('renders correctly with props border and borderWidth', () => {
    const tree = renderer.create(
      <IconSvg border borderWidth={2} />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('renders correctly with props border and rounded', () => {
    const tree = renderer.create(
      <IconSvg border rounded />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('renders correctly with props border, borderWidth, rounded', () => {
    const tree = renderer.create(
      <IconSvg
        border
        rounded
        borderWidth={2}
      />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('renders correctly with prop size=small', () => {
    const tree = renderer.create(
      <IconSvg size="small" />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('renders correctly with prop size=large', () => {
    const tree = renderer.create(
      <IconSvg size="large" />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('renders correctly with prop size=xlarge', () => {
    const tree = renderer.create(
      <IconSvg size="xlarge" />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('renders correctly with prop colorScheme=light', () => {
    const tree = renderer.create(
      <IconSvg
        colorScheme="light"
      />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('renders correctly with prop color', () => {
    const tree = renderer.create(
      <IconSvg color="#02b4d4" />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('renders correctly with prop backgroundColor', () => {
    const tree = renderer.create(
      <IconSvg backgroundColor="#e81f82" />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('renders correctly with props backgroundColor and color', () => {
    const tree = renderer.create(
      <IconSvg
        backgroundColor="#e81f82"
        color="#02b4d4"
      />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('renders correctly with prop flip=horizontal', () => {
    const tree = renderer.create(
      <IconSvg flip="horizontal" />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('renders correctly with prop flip=vertical', () => {
    const tree = renderer.create(
      <IconSvg flip="vertical" />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('renders correctly with prop rotate', () => {
    const tree = renderer.create(
      <IconSvg rotate={45} />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
