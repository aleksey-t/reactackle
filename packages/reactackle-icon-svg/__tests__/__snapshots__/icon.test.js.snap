// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconSvg/> renders correctly with border prop 1`] = `
<div
  className="IconSvgStyled__IconSvgStyled-iCtyLo cRTIbe"
  color=""
  rotate={0}
  size="normal"
>
  <svg
    preserveAspectRatio="xMidYMid meet"
    viewBox="0 0 24 24"
  >
    <path
      d="M19,5.5c0-0.1,0-0.1,0-0.2c-0.1-0.1-0.1-0.2-0.3-0.3c-0.1,0-0.1,0-0.2,0h-10C8.4,5,8.4,5,8.3,5c-0.1,0-0.1,0.1-0.2,0.1l-3,3 c0,0-0.1,0.1-0.1,0.2c0,0.1,0,0.1,0,0.2v10C5,18.8,5.2,19,5.5,19h10c0.1,0,0.1,0,0.2,0c0.1,0,0.1-0.1,0.2-0.1l3-3 c0,0,0.1-0.1,0.1-0.2c0-0.1,0-0.1,0-0.2V5.5z M16,16h1.3L16,17.3V16z M18,15h-2V8.7l2-2V15z M9,9h6v6H9V9z M8,8H6.7L8,6.7V8z M9,6 h8.3l-2,2H9V6z M6,9h2v6.5C8,15.8,8.2,16,8.5,16H15v2H6V9z"
    />
  </svg>
</div>
`;

exports[`<IconSvg/> renders correctly with children prop 1`] = `
<div
  className="IconSvgStyled__IconSvgStyled-iCtyLo jTBtSb"
  color=""
  rotate={0}
  size="normal"
>
  <svg
    preserveAspectRatio="xMidYMid meet"
    viewBox="0 0 24 24"
  >
    <path
      d="M7,10l5,5l5-5H7z"
    />
  </svg>
</div>
`;

exports[`<IconSvg/> renders correctly with default props 1`] = `
<div
  className="IconSvgStyled__IconSvgStyled-iCtyLo jTBtSb"
  color=""
  rotate={0}
  size="normal"
>
  <svg
    preserveAspectRatio="xMidYMid meet"
    viewBox="0 0 24 24"
  >
    <path
      d="M19,5.5c0-0.1,0-0.1,0-0.2c-0.1-0.1-0.1-0.2-0.3-0.3c-0.1,0-0.1,0-0.2,0h-10C8.4,5,8.4,5,8.3,5c-0.1,0-0.1,0.1-0.2,0.1l-3,3 c0,0-0.1,0.1-0.1,0.2c0,0.1,0,0.1,0,0.2v10C5,18.8,5.2,19,5.5,19h10c0.1,0,0.1,0,0.2,0c0.1,0,0.1-0.1,0.2-0.1l3-3 c0,0,0.1-0.1,0.1-0.2c0-0.1,0-0.1,0-0.2V5.5z M16,16h1.3L16,17.3V16z M18,15h-2V8.7l2-2V15z M9,9h6v6H9V9z M8,8H6.7L8,6.7V8z M9,6 h8.3l-2,2H9V6z M6,9h2v6.5C8,15.8,8.2,16,8.5,16H15v2H6V9z"
    />
  </svg>
</div>
`;

exports[`<IconSvg/> renders correctly with prop backgroundColor 1`] = `
<div
  className="IconSvgStyled__IconSvgStyled-iCtyLo ikOmjR"
  color=""
  rotate={0}
  size="normal"
>
  <svg
    preserveAspectRatio="xMidYMid meet"
    viewBox="0 0 24 24"
  >
    <path
      d="M19,5.5c0-0.1,0-0.1,0-0.2c-0.1-0.1-0.1-0.2-0.3-0.3c-0.1,0-0.1,0-0.2,0h-10C8.4,5,8.4,5,8.3,5c-0.1,0-0.1,0.1-0.2,0.1l-3,3 c0,0-0.1,0.1-0.1,0.2c0,0.1,0,0.1,0,0.2v10C5,18.8,5.2,19,5.5,19h10c0.1,0,0.1,0,0.2,0c0.1,0,0.1-0.1,0.2-0.1l3-3 c0,0,0.1-0.1,0.1-0.2c0-0.1,0-0.1,0-0.2V5.5z M16,16h1.3L16,17.3V16z M18,15h-2V8.7l2-2V15z M9,9h6v6H9V9z M8,8H6.7L8,6.7V8z M9,6 h8.3l-2,2H9V6z M6,9h2v6.5C8,15.8,8.2,16,8.5,16H15v2H6V9z"
    />
  </svg>
</div>
`;

exports[`<IconSvg/> renders correctly with prop color 1`] = `
<div
  className="IconSvgStyled__IconSvgStyled-iCtyLo cxNzEJ"
  color=""
  rotate={0}
  size="normal"
>
  <svg
    preserveAspectRatio="xMidYMid meet"
    viewBox="0 0 24 24"
  >
    <path
      d="M19,5.5c0-0.1,0-0.1,0-0.2c-0.1-0.1-0.1-0.2-0.3-0.3c-0.1,0-0.1,0-0.2,0h-10C8.4,5,8.4,5,8.3,5c-0.1,0-0.1,0.1-0.2,0.1l-3,3 c0,0-0.1,0.1-0.1,0.2c0,0.1,0,0.1,0,0.2v10C5,18.8,5.2,19,5.5,19h10c0.1,0,0.1,0,0.2,0c0.1,0,0.1-0.1,0.2-0.1l3-3 c0,0,0.1-0.1,0.1-0.2c0-0.1,0-0.1,0-0.2V5.5z M16,16h1.3L16,17.3V16z M18,15h-2V8.7l2-2V15z M9,9h6v6H9V9z M8,8H6.7L8,6.7V8z M9,6 h8.3l-2,2H9V6z M6,9h2v6.5C8,15.8,8.2,16,8.5,16H15v2H6V9z"
    />
  </svg>
</div>
`;

exports[`<IconSvg/> renders correctly with prop colorScheme=light 1`] = `
<div
  className="IconSvgStyled__IconSvgStyled-iCtyLo ibIFIX"
  color=""
  rotate={0}
  size="normal"
>
  <svg
    preserveAspectRatio="xMidYMid meet"
    viewBox="0 0 24 24"
  >
    <path
      d="M19,5.5c0-0.1,0-0.1,0-0.2c-0.1-0.1-0.1-0.2-0.3-0.3c-0.1,0-0.1,0-0.2,0h-10C8.4,5,8.4,5,8.3,5c-0.1,0-0.1,0.1-0.2,0.1l-3,3 c0,0-0.1,0.1-0.1,0.2c0,0.1,0,0.1,0,0.2v10C5,18.8,5.2,19,5.5,19h10c0.1,0,0.1,0,0.2,0c0.1,0,0.1-0.1,0.2-0.1l3-3 c0,0,0.1-0.1,0.1-0.2c0-0.1,0-0.1,0-0.2V5.5z M16,16h1.3L16,17.3V16z M18,15h-2V8.7l2-2V15z M9,9h6v6H9V9z M8,8H6.7L8,6.7V8z M9,6 h8.3l-2,2H9V6z M6,9h2v6.5C8,15.8,8.2,16,8.5,16H15v2H6V9z"
    />
  </svg>
</div>
`;

exports[`<IconSvg/> renders correctly with prop flip=horizontal 1`] = `
<div
  className="IconSvgStyled__IconSvgStyled-iCtyLo bpDoAU"
  color=""
  rotate={0}
  size="normal"
>
  <svg
    preserveAspectRatio="xMidYMid meet"
    viewBox="0 0 24 24"
  >
    <path
      d="M19,5.5c0-0.1,0-0.1,0-0.2c-0.1-0.1-0.1-0.2-0.3-0.3c-0.1,0-0.1,0-0.2,0h-10C8.4,5,8.4,5,8.3,5c-0.1,0-0.1,0.1-0.2,0.1l-3,3 c0,0-0.1,0.1-0.1,0.2c0,0.1,0,0.1,0,0.2v10C5,18.8,5.2,19,5.5,19h10c0.1,0,0.1,0,0.2,0c0.1,0,0.1-0.1,0.2-0.1l3-3 c0,0,0.1-0.1,0.1-0.2c0-0.1,0-0.1,0-0.2V5.5z M16,16h1.3L16,17.3V16z M18,15h-2V8.7l2-2V15z M9,9h6v6H9V9z M8,8H6.7L8,6.7V8z M9,6 h8.3l-2,2H9V6z M6,9h2v6.5C8,15.8,8.2,16,8.5,16H15v2H6V9z"
    />
  </svg>
</div>
`;

exports[`<IconSvg/> renders correctly with prop flip=vertical 1`] = `
<div
  className="IconSvgStyled__IconSvgStyled-iCtyLo IIQTa"
  color=""
  rotate={0}
  size="normal"
>
  <svg
    preserveAspectRatio="xMidYMid meet"
    viewBox="0 0 24 24"
  >
    <path
      d="M19,5.5c0-0.1,0-0.1,0-0.2c-0.1-0.1-0.1-0.2-0.3-0.3c-0.1,0-0.1,0-0.2,0h-10C8.4,5,8.4,5,8.3,5c-0.1,0-0.1,0.1-0.2,0.1l-3,3 c0,0-0.1,0.1-0.1,0.2c0,0.1,0,0.1,0,0.2v10C5,18.8,5.2,19,5.5,19h10c0.1,0,0.1,0,0.2,0c0.1,0,0.1-0.1,0.2-0.1l3-3 c0,0,0.1-0.1,0.1-0.2c0-0.1,0-0.1,0-0.2V5.5z M16,16h1.3L16,17.3V16z M18,15h-2V8.7l2-2V15z M9,9h6v6H9V9z M8,8H6.7L8,6.7V8z M9,6 h8.3l-2,2H9V6z M6,9h2v6.5C8,15.8,8.2,16,8.5,16H15v2H6V9z"
    />
  </svg>
</div>
`;

exports[`<IconSvg/> renders correctly with prop rotate 1`] = `
<div
  className="IconSvgStyled__IconSvgStyled-iCtyLo fhRFOp"
  color=""
  rotate={45}
  size="normal"
>
  <svg
    preserveAspectRatio="xMidYMid meet"
    viewBox="0 0 24 24"
  >
    <path
      d="M19,5.5c0-0.1,0-0.1,0-0.2c-0.1-0.1-0.1-0.2-0.3-0.3c-0.1,0-0.1,0-0.2,0h-10C8.4,5,8.4,5,8.3,5c-0.1,0-0.1,0.1-0.2,0.1l-3,3 c0,0-0.1,0.1-0.1,0.2c0,0.1,0,0.1,0,0.2v10C5,18.8,5.2,19,5.5,19h10c0.1,0,0.1,0,0.2,0c0.1,0,0.1-0.1,0.2-0.1l3-3 c0,0,0.1-0.1,0.1-0.2c0-0.1,0-0.1,0-0.2V5.5z M16,16h1.3L16,17.3V16z M18,15h-2V8.7l2-2V15z M9,9h6v6H9V9z M8,8H6.7L8,6.7V8z M9,6 h8.3l-2,2H9V6z M6,9h2v6.5C8,15.8,8.2,16,8.5,16H15v2H6V9z"
    />
  </svg>
</div>
`;

exports[`<IconSvg/> renders correctly with prop size=large 1`] = `
<div
  className="IconSvgStyled__IconSvgStyled-iCtyLo eaeAsm"
  color=""
  rotate={0}
  size="normal"
>
  <svg
    preserveAspectRatio="xMidYMid meet"
    viewBox="0 0 24 24"
  >
    <path
      d="M19,5.5c0-0.1,0-0.1,0-0.2c-0.1-0.1-0.1-0.2-0.3-0.3c-0.1,0-0.1,0-0.2,0h-10C8.4,5,8.4,5,8.3,5c-0.1,0-0.1,0.1-0.2,0.1l-3,3 c0,0-0.1,0.1-0.1,0.2c0,0.1,0,0.1,0,0.2v10C5,18.8,5.2,19,5.5,19h10c0.1,0,0.1,0,0.2,0c0.1,0,0.1-0.1,0.2-0.1l3-3 c0,0,0.1-0.1,0.1-0.2c0-0.1,0-0.1,0-0.2V5.5z M16,16h1.3L16,17.3V16z M18,15h-2V8.7l2-2V15z M9,9h6v6H9V9z M8,8H6.7L8,6.7V8z M9,6 h8.3l-2,2H9V6z M6,9h2v6.5C8,15.8,8.2,16,8.5,16H15v2H6V9z"
    />
  </svg>
</div>
`;

exports[`<IconSvg/> renders correctly with prop size=small 1`] = `
<div
  className="IconSvgStyled__IconSvgStyled-iCtyLo fkRTgE"
  color=""
  rotate={0}
  size="normal"
>
  <svg
    preserveAspectRatio="xMidYMid meet"
    viewBox="0 0 24 24"
  >
    <path
      d="M19,5.5c0-0.1,0-0.1,0-0.2c-0.1-0.1-0.1-0.2-0.3-0.3c-0.1,0-0.1,0-0.2,0h-10C8.4,5,8.4,5,8.3,5c-0.1,0-0.1,0.1-0.2,0.1l-3,3 c0,0-0.1,0.1-0.1,0.2c0,0.1,0,0.1,0,0.2v10C5,18.8,5.2,19,5.5,19h10c0.1,0,0.1,0,0.2,0c0.1,0,0.1-0.1,0.2-0.1l3-3 c0,0,0.1-0.1,0.1-0.2c0-0.1,0-0.1,0-0.2V5.5z M16,16h1.3L16,17.3V16z M18,15h-2V8.7l2-2V15z M9,9h6v6H9V9z M8,8H6.7L8,6.7V8z M9,6 h8.3l-2,2H9V6z M6,9h2v6.5C8,15.8,8.2,16,8.5,16H15v2H6V9z"
    />
  </svg>
</div>
`;

exports[`<IconSvg/> renders correctly with prop size=xlarge 1`] = `
<div
  className="IconSvgStyled__IconSvgStyled-iCtyLo eVtvkX"
  color=""
  rotate={0}
  size="normal"
>
  <svg
    preserveAspectRatio="xMidYMid meet"
    viewBox="0 0 24 24"
  >
    <path
      d="M19,5.5c0-0.1,0-0.1,0-0.2c-0.1-0.1-0.1-0.2-0.3-0.3c-0.1,0-0.1,0-0.2,0h-10C8.4,5,8.4,5,8.3,5c-0.1,0-0.1,0.1-0.2,0.1l-3,3 c0,0-0.1,0.1-0.1,0.2c0,0.1,0,0.1,0,0.2v10C5,18.8,5.2,19,5.5,19h10c0.1,0,0.1,0,0.2,0c0.1,0,0.1-0.1,0.2-0.1l3-3 c0,0,0.1-0.1,0.1-0.2c0-0.1,0-0.1,0-0.2V5.5z M16,16h1.3L16,17.3V16z M18,15h-2V8.7l2-2V15z M9,9h6v6H9V9z M8,8H6.7L8,6.7V8z M9,6 h8.3l-2,2H9V6z M6,9h2v6.5C8,15.8,8.2,16,8.5,16H15v2H6V9z"
    />
  </svg>
</div>
`;

exports[`<IconSvg/> renders correctly with props backgroundColor and color 1`] = `
<div
  className="IconSvgStyled__IconSvgStyled-iCtyLo bgGOOb"
  color=""
  rotate={0}
  size="normal"
>
  <svg
    preserveAspectRatio="xMidYMid meet"
    viewBox="0 0 24 24"
  >
    <path
      d="M19,5.5c0-0.1,0-0.1,0-0.2c-0.1-0.1-0.1-0.2-0.3-0.3c-0.1,0-0.1,0-0.2,0h-10C8.4,5,8.4,5,8.3,5c-0.1,0-0.1,0.1-0.2,0.1l-3,3 c0,0-0.1,0.1-0.1,0.2c0,0.1,0,0.1,0,0.2v10C5,18.8,5.2,19,5.5,19h10c0.1,0,0.1,0,0.2,0c0.1,0,0.1-0.1,0.2-0.1l3-3 c0,0,0.1-0.1,0.1-0.2c0-0.1,0-0.1,0-0.2V5.5z M16,16h1.3L16,17.3V16z M18,15h-2V8.7l2-2V15z M9,9h6v6H9V9z M8,8H6.7L8,6.7V8z M9,6 h8.3l-2,2H9V6z M6,9h2v6.5C8,15.8,8.2,16,8.5,16H15v2H6V9z"
    />
  </svg>
</div>
`;

exports[`<IconSvg/> renders correctly with props border and borderWidth 1`] = `
<div
  className="IconSvgStyled__IconSvgStyled-iCtyLo blQAut"
  color=""
  rotate={0}
  size="normal"
>
  <svg
    preserveAspectRatio="xMidYMid meet"
    viewBox="0 0 24 24"
  >
    <path
      d="M19,5.5c0-0.1,0-0.1,0-0.2c-0.1-0.1-0.1-0.2-0.3-0.3c-0.1,0-0.1,0-0.2,0h-10C8.4,5,8.4,5,8.3,5c-0.1,0-0.1,0.1-0.2,0.1l-3,3 c0,0-0.1,0.1-0.1,0.2c0,0.1,0,0.1,0,0.2v10C5,18.8,5.2,19,5.5,19h10c0.1,0,0.1,0,0.2,0c0.1,0,0.1-0.1,0.2-0.1l3-3 c0,0,0.1-0.1,0.1-0.2c0-0.1,0-0.1,0-0.2V5.5z M16,16h1.3L16,17.3V16z M18,15h-2V8.7l2-2V15z M9,9h6v6H9V9z M8,8H6.7L8,6.7V8z M9,6 h8.3l-2,2H9V6z M6,9h2v6.5C8,15.8,8.2,16,8.5,16H15v2H6V9z"
    />
  </svg>
</div>
`;

exports[`<IconSvg/> renders correctly with props border and rounded 1`] = `
<div
  className="IconSvgStyled__IconSvgStyled-iCtyLo cITZPs"
  color=""
  rotate={0}
  size="normal"
>
  <svg
    preserveAspectRatio="xMidYMid meet"
    viewBox="0 0 24 24"
  >
    <path
      d="M19,5.5c0-0.1,0-0.1,0-0.2c-0.1-0.1-0.1-0.2-0.3-0.3c-0.1,0-0.1,0-0.2,0h-10C8.4,5,8.4,5,8.3,5c-0.1,0-0.1,0.1-0.2,0.1l-3,3 c0,0-0.1,0.1-0.1,0.2c0,0.1,0,0.1,0,0.2v10C5,18.8,5.2,19,5.5,19h10c0.1,0,0.1,0,0.2,0c0.1,0,0.1-0.1,0.2-0.1l3-3 c0,0,0.1-0.1,0.1-0.2c0-0.1,0-0.1,0-0.2V5.5z M16,16h1.3L16,17.3V16z M18,15h-2V8.7l2-2V15z M9,9h6v6H9V9z M8,8H6.7L8,6.7V8z M9,6 h8.3l-2,2H9V6z M6,9h2v6.5C8,15.8,8.2,16,8.5,16H15v2H6V9z"
    />
  </svg>
</div>
`;

exports[`<IconSvg/> renders correctly with props border, borderWidth, rounded 1`] = `
<div
  className="IconSvgStyled__IconSvgStyled-iCtyLo kpqvRP"
  color=""
  rotate={0}
  size="normal"
>
  <svg
    preserveAspectRatio="xMidYMid meet"
    viewBox="0 0 24 24"
  >
    <path
      d="M19,5.5c0-0.1,0-0.1,0-0.2c-0.1-0.1-0.1-0.2-0.3-0.3c-0.1,0-0.1,0-0.2,0h-10C8.4,5,8.4,5,8.3,5c-0.1,0-0.1,0.1-0.2,0.1l-3,3 c0,0-0.1,0.1-0.1,0.2c0,0.1,0,0.1,0,0.2v10C5,18.8,5.2,19,5.5,19h10c0.1,0,0.1,0,0.2,0c0.1,0,0.1-0.1,0.2-0.1l3-3 c0,0,0.1-0.1,0.1-0.2c0-0.1,0-0.1,0-0.2V5.5z M16,16h1.3L16,17.3V16z M18,15h-2V8.7l2-2V15z M9,9h6v6H9V9z M8,8H6.7L8,6.7V8z M9,6 h8.3l-2,2H9V6z M6,9h2v6.5C8,15.8,8.2,16,8.5,16H15v2H6V9z"
    />
  </svg>
</div>
`;

exports[`<IconSvg/> renders correctly with src prop 1`] = `
<div
  className="IconSvgStyled__IconSvgStyled-iCtyLo jTBtSb"
  color=""
  rotate={0}
  size="normal"
>
  <svg
    preserveAspectRatio="xMidYMid meet"
    viewBox="0 0 24 24"
  >
    <path
      d="M7,10l5,5l5-5H7z"
    />
  </svg>
</div>
`;
